Introduce editable table for creating score distributions

jorg-vr opened the pull request:
This pull request introduces [jspreadsheet-ce](https://github.com/jspreadsheet/ce) as table editor to manage score items for an evaluation.

This way, copying, reusing and even initially filling it out all go a lot faster. The excel like layout should be easy to interact with for most users.

It also allows to prepare score items in advance in a spreadsheet program and pasting it here directly.

Reordering is partially supported. Doing when some solutions are already evaluated will force you to redo those evaluations.

https://github.com/user-attachments/assets/4af53f70-c816-4b0d-849d-e59438b36765

Darkmode is also supported
![image](https://github.com/user-attachments/assets/00f3fa50-f779-4a4e-ba78-fbab5be8d0eb)

- [x] Tests were added
- [ ] Documentation update can be found at dodona-edu/dodona-edu.github.io#

Closes #4940 and closes #2713

bmesuere reviewed (changes requested):
First of all, very nice work!

 - [ ] We semi-discussed that clicking on the table would also trigger edit mode. Is there a reason why this isn't implemented?

- [ ] I found an edge-case when entering numbers using `,` instead of `.`. If I enter `1,5` as a score, I don't get an error that I need to enter a number, but saving also seems to fail silently. I think we should both accept `1.5` as well as `1,5`.

- [ ] Do we have control over the editor? If so, for the header row and col, maybe try to reduce the font size a little and make it bold to be more in line with the actual table.

- [ ] The Maximum column is centered, but jumps to left aligned when editing. Why not alway align left?

- [ ] The tooltip of the Maximum column mentions steps of 0.25 but I think we allow steps of 0.1?

- [ ] If I remember correctly, you demoed copying entire rows. I can't seem to reproduce this. I can select a row, press ctrl+c and the row gets a dashed line, but pasting doesn't do anything.

- [ ] In the form to add comments to submissions, we didn't give the cancel button an outline. I would also not do that here. This will also increase the visual padding between the two buttons.

jorg-vr commented:
> The tooltip of the Maximum column mentions steps of 0.25 but I think we allow steps of 0.1?

We allow steps of 0.01. But when giving grades, the number increase/decrease buttons take steps of 0.25. So It might be advised to also use steps of 0.25 for your max.

I am open to removing the from the tooltip, enforcing the 0.25 or simply leaving as is.

jorg-vr commented:
> If I remember correctly, you demoed copying entire rows. I can't seem to reproduce this. I can select a row, press ctrl+c and the row gets a dashed line, but pasting doesn't do anything.

In Firefox everything works as expected

In chrome `ctrl+v` doesn't always work as expected. Copying using `ctrl-c` or cut using `ctrl-x` always works correctly.
After copying or cutting, pasting doesn't work using `ctrl+v`
What does work:
- pasting works using right click paste
- or after first focusing an input field (eg after double clicking a cell) and then unfocusing and simply doing `ctrl-v`

`ctrl+v`  also work if you haven't just copied something from a js-spreadsheet on the same page (eg copied from a google sheet)

bmesuere reviewed (changes requested):
- [x] When hovering over the tables (which is editable on click), set the cursor to `text` to hint at editability
- [x] The flow to add a score item is not very clear. It becomes clear when you are in edit mode, but it is not clear that you must enter edit mode to add an item. I would at least add a button/link to the empty state. In addition, instead of just a pencil icon, we could maybe add/replace it with an "edit scores" button
- [ ] Likewise, it is not clear/easy to remove a row. I think the only way is to right-click a row header?

jorg-vr commented on app/assets/javascripts/components/input_table.ts:
```diff
@@ -0,0 +1,304 @@
+            }
+            // Check if maximum is a positive number < 1000
+            // we use a regex instead of parseFloat because parseFloat is too lenient
+            if (! /^\d{1,3}(\.\d*)?$/.test(item.maximum) || parseFloat(item.maximum) <= 0) {
```
explanation is wrong, edit suggestion is an improvement

bmesuere reviewed (changes requested):
Row removal is a lot clearer now 👍
A few more comments:
- the row remove button has the material mouseover circle, but this is a bit strange in the table which is style differently (and part of the circle is cut off)
- the visibility checkbox in the table is aligned badly (bottom-left)
  ![image](https://github.com/user-attachments/assets/cff0aa0c-9614-4c8a-9a9a-50beb7d54cab)
- I apparently forgot to write down my comment on the max score visibility. The toggle there looks very strange
- I'm not totally convinced that swapping out the pencil for "edit" fixed the problem. To me, there's a disconnect between the button and the table however I have no clear suggestion on how to fix this.
  ![image](https://github.com/user-attachments/assets/263fbd38-993f-4314-b700-64c22b4c3e1d) 
- not changed in this PR but strange when I think about it: the bottom row now says maximum score but should this be total score?

Sorry for nitpicking!

jorg-vr commented:
On the delete button => I made a new simpler button type where hovering is indicated by opacity reduction

For the edit button => I removed the edit button completely in this version. Instead I added some color indication to table cells when hovering them, making it more clear that they are editable.

https://github.com/user-attachments/assets/4d7525b2-ab62-404a-adb9-9cbcba62ae7b

jorg-vr commented:
- [ ] delete button => color darker
- [ ] Always show 'add score items'
